src/parse_fc.c

```diff
@@ -243,22 +251,52 @@ struct policy_node *parse_fc_file(const char *filename, const struct string_list
 	struct policy_node *cur = head;
 
 	char *line = NULL;
+	char *ifdef_condition = NULL;
+	char *ifndef_condition = NULL;
+	char *token = NULL;
+	struct conditional_data *conditional = NULL;
 
 	ssize_t len_read = 0;
 	size_t buf_len = 0;
 	unsigned int lineno = 0;
+	bool is_within_ifdef = false;
+	bool is_within_ifndef = false;
 	while ((len_read = getline(&line, &buf_len, fd)) != -1) {
 		lineno++;
 		if (len_read <= 1 || line[0] == '#') {
 			continue;
-		}
-		// Skip over m4 constructs
-		if (strncmp(line, "ifdef", 5) == 0 ||
-		    strncmp(line, "ifndef", 6) == 0 ||
-		    strncmp(line, "')", 2) == 0 ||
-		    strncmp(line, "', `", 4) == 0 ||
-		    strncmp(line, "',`", 3) == 0) {
-
+		} else if (!strncmp(line, "ifdef", 5)) {
+			is_within_ifdef = true;
+			token = strtok(line, "`");
+			if (token) {
+				token = strtok(NULL, "`");
+				token = strtok(token, "'");
+				ifdef_condition = malloc(strlen(token) + 1);
+				ifdef_condition = strdup(token);
+			}
+			continue;
+		} else if (!strncmp(line, "ifndef", 6)) {
+			is_within_ifndef = true;
+			token = strtok(line, "`");
+			if (token) {
+				token = strtok(NULL, "`");
+				token = strtok(token, "'");
+				ifndef_condition = malloc(strlen(token) + 1);
+				ifndef_condition = strdup(token);
+			}
+			continue;
+		} else if (!strncmp(line, "')", 2)) {
```

Comment:
The lack of durability here scares me.  Ending an ifdef block with eg `' )` (note space) would be legal, and missed by this.

The durable solution is probably a fairly large and high risk change, so maybe for now, just a TODO comment noting the issue?  SELint does generally assume reasonable style, but this one seems pretty annoying if it were to hit.